R/plot.R

```diff
@@ -942,13 +942,13 @@ plot_pit <- function(pit,
 #' @description
 #' Visualise Where Forecasts Are Available
 #' @inheritParams print.scoringutils_check
-#' @param x an S3 object of class "scoringutils_available_forecasts"
-#' as produced by [available_forecasts()]
-#' @param yvar character vector of length one that denotes the name of the column
+#' @param forecast_counts a data.table (or similar) with forecast counts
+#' as produced by [get_forecast_counts()]
+#' @param y character vector of length one that denotes the name of the column
 #' to appear on the y-axis of the plot. Default is "model".
-#' @param xvar character vector of length one that denotes the name of the column
-#' to appear on the x-axis of the plot. Default is "forecast_date".
-#' @param make_xvar_factor logical (default is TRUE). Whether or not to convert
+#' @param x character vector of length one that denotes the name of the column
+#' to appear on the x-axis of the plot.
+#' @param make_x_factor logical (default is TRUE). Whether or not to convert
```

Comment:
what about `x_as_factor`? TBH I am not sure why we need this vs getting the user to do it themselves?

Reply:
Extrapolating from me as a user I found it quite useful.. My impression was you'd almost always want this to be a factor and that's why I included it in the function

```
get_forecast_counts(example_quantile, by = c("forecast_date", "target_type")) |> 
   plot_forecast_counts(y = "target_type", x = "forecast_date", make_x_factor = FALSE)
```
<img width="720" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/93b16317-c5ca-495e-a670-06be81e76aa0">

vs. 

```
get_forecast_counts(example_quantile, by = c("forecast_date", "target_type")) |> 
   plot_forecast_counts(y = "target_type", x = "forecast_date", make_x_factor = TRUE)
```
<img width="719" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/661cda7b-4448-441f-8335-c1e655befdc1">

Reply:
I don't feel that strongly but would vote to leave it to the user to decide/manage vs helping them convert a factor.